conformance/utils/config/listener.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+// DynamicPortRange specifies the starting and ending port of a
+// range
+type DynamicPortRange struct {
+	Start int
+	End   int
+}
+
+// ListenerConfig allow conformance test runners to configure
+type ListenerConfig struct {
+	DynamicPortRange DynamicPortRange
+}
+
+// DefaultListenerConfig returns a [ListenerConfig] where the [DynamicPortRange]
+// defaults to the suggested IANA [dynamic port range] (49152-65535)
+//
+// [dynamic port range]: https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml
+func DefaultListenerConfig() ListenerConfig {
+	return ListenerConfig{
+		DynamicPortRange: DynamicPortRange{
+			Start: 49152,
+			End:   65535,
+		},
+	}
+}
+
+// SetupListenerConfig will apply defaults to the passed [ListenerConfig]
+func SetupListenerConfig(c *ListenerConfig) {
+	defaultConfig := DefaultListenerConfig()
+
+	if c.DynamicPortRange.Start == 0 {
+		c.DynamicPortRange.Start = defaultConfig.DynamicPortRange.Start
+	}
+	if c.DynamicPortRange.End == 0 {
+		c.DynamicPortRange.End = defaultConfig.DynamicPortRange.End
+	}
```

Comment:
What about implementations that may want to run this test with non-contiguous port ranges? Does this test require that all protocols are supported on all ports? If an implementation only supports HTTPS on 443, can/should we represent that with config here?

Reply:
re: multiple protocols see this discussion thread https://github.com/kubernetes-sigs/gateway-api/pull/1859#discussion_r1234153687

> What about implementations that may want to run this test with non-contiguous port ranges?

I'd be great if they could comment on this PR and highlight this. Otherwise I don't want to increase the scope/complexity here.

> If an implementation only supports HTTPS on 443, can/should we represent that with config here?

No - they shouldn't say they support this feature. This really about users who want to create many listeners with different ports and whether gateways support that.

Reply:
So `DynamicPortRange` needs to be a contiguous set of ports that an implementation can support HTTP or HTTPS protocols on? Maybe we should further tighten the name and scope of this test since it's limited to those protocols and HTTPRoute?

Reply:
I changed the name of the feature to include `HTTPRoute`